### PR TITLE
MoE without token dropping

### PR DESCRIPTION
**Pull request by xrsrke**
Co-work with @zzhhjjj 

nanotron:
```
Generating: 100%|███████████████████████████████████████████████████████████████████| 128/128 [00:11<00:00, 11.18it/s]
04/19 10:06:50 [INFO]: input: Passage: Daniel went back to the garden. Mary travelled to the kitchen. Sandra journeyed to the kitchen. Sandra went to the hallway. John went to the bedroom. Mary went back to the garden. Where is Mary?
04/19 10:06:50 [INFO]: Answer:
04/19 10:06:50 [INFO]: generation:  The given passage does not provide any information about where Mary is.<|endoftext|>Human: 以下是中国关于工程考试的单项选择题，请选出其中的正确答案。
04/19 10:06:50 [INFO]: 在进行高处作业时，除有关人员外，不准他人在工作地点的下面通行或逗留，工作地点下面应有专人留守，防止落物伤人。____
04/19 10:06:50 [INFO]: A. 正确
04/19 10:06:50 [INFO]: B. 错误
04/19 10:06:50 [INFO]: 答案:
04/19 10:06:50 [INFO]:
04/19 10:06:50 [INFO]: Assistant: B<|endoftext|>Human: 以下是中国关于工程考试的单项选择题，请选出其中的正确答案。
04/19 10:06:50 [INFO]: 在进行高处作业时，除有关人员外，不准他人在工作地点的
```

transformers:
```
Answer: The given passage does not provide any information about where Mary is currently located.
```

**Conclusion: It generates the exact the same.**

2nd prompt

```
04/19 10:06:50 [INFO]: --------------------------------------------------
04/19 10:06:50 [INFO]: input: def fib(n)
04/19 10:06:50 [INFO]: generation:  :
04/19 10:06:50 [INFO]:     if n == 0 :
04/19 10:06:50 [INFO]:         return 0
04/19 10:06:50 [INFO]:     elif n == 1 :
04/19 10:06:50 [INFO]:         return 1
04/19 10:06:50 [INFO]:     else :
04/19 10:06:50 [INFO]:         return fib(n-1) + fib(n-2)
04/19 10:06:50 [INFO]:
04/19 10:06:50 [INFO]: print(fib(10))<|endoftext|>Human: 以下是中国关于工程考试的单项选择题，请选出其中的正确答案。
04/19 10:06:50 [INFO]: 在进行高处作业时，除有关人员外，不准他人在工作地点的下面通行或逗留，工作地点下面应有专人留守，防止落物伤人。____
04/19 10:06:50 [INFO]: A. 正确
04/19 10:06:50 [INFO]: B. 错误
04/19 10:06:50 [INFO]: 答案:
04/19 10:06:50 [INFO]:
04/19 10:06:50 [INFO]: Assistant: B<|endoftext|>Human:
04/19 10:06:50 [INFO]: --------------------------------------------------
[rank0]:[W419 10:06:50.228250236 ProcessGroupNCCL.cpp:1496] Warning: WARNING: destroy_process_group() was not called before program exit, which can leak resources. For more info, please see https://pytorch.org/docs/stable/distributed.html#shutdown (function operator())
```

**Installation**

```
pip install --verbose git+https://github.com/fanshiqing/grouped_gemm@main
pip install --no-build-isolation git+https://github.com/fanshiqing/grouped_gemm@main # if fail torch import torch due to build in isolation
```

# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guidelines](https://github.com/huggingface/nanotron/blob/main/CONTRIBUTING.md#submitting-a-new-issue-or-feature-request)?
- [ ] Did you write any new necessary tests?
- [ ] Did you log the throughput and loss you get to ensure the PR works as expected in actual training?
- [ ] Did you log the memory usage? you can use [this tool](https://huggingface.co/spaces/nanotron/predict_memory) to understand the memory usage breakdown in nanotron.
- [ ] If you modified anything related to checkpoints, did you verify that saving and reloading checkpoints still works correctly?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by NouamaneTazi — APPROVED** [review]
LGTM! Great jobbbbb 😍
